### PR TITLE
add base64 and dict_merge extension

### PR DESCRIPTION
**Pull request by gebinic**
Added an extension which encodes and decodes a string. Supported unix and windows systems.

EDIT: add dict_merge extension

### CODE REVIEW

**Comment by nicks** [comment]
thanks for the contribution!

i'm fine with the `base64` extension

i'm nervous about having the `dict_merge` extension in the main extensions repo, because:
- starlark already has a dictionary union operator (see: https://github.com/bazelbuild/starlark/blob/master/spec.md#dictionaries)
- historically, deep merge has been pretty controversial, because it's hard to make the behavior intuitive

**Comment by gebinic** [comment]
We distinguish between `deploy_config.yml` ("global" configuration file) and `local_config.yml` (git ignored file, only local user configuration file). Due to our configuration model, it is extremely difficult to merge these configuration files manually, among other things with the dictionary union operator. Therefore, we decided to introduce the `dict_merge` function, which simply merges all configurations deeply. I think this could be a use case that also affects other tilt users.

File: base64/Tiltfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+def encode_base64(content):
+    command_value = 'echo {} | base64'.format(content)
```
Comment (remram44) [inline base64/Tiltfile:2]: With longer strings, base64 will insert newlines every 76 characters. `base64 -w 0` avoids that.

**Comment by nicks** [comment]
i think we'd accept a pr for just the `base64` but wouldn't accept `dict_merge`, for the reasons stated above. closing this pr, but feel free to repopen for the `base64` part

**Pull request closed without merging**